webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
+    } catch (Exception e) {
+      throw new JobExecutionException(e);
+    }
+  }
+
+  @Override
+  public void onSuccess(JobExecutionContext context) {
+    logger.info(
+        "Third-Party Sync succeeded for repository {}.", scheduledJob.getRepository().getName());
+
+    // Resolve PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              try {
+                pd.resolveIncident(scheduledJob.getId());
+              } catch (PagerDutyException e) {
+                logger.error(
+                    "Couldn't send resolve PagerDuty notification for successful third party sync of repository: '{}' -",
+                    scheduledJob.getRepository().getName(),
+                    e);
+              }
+            });
+  }
+
+  @Override
+  public void onFailure(JobExecutionContext context, JobExecutionException jobException) {
+    logger.error(
+        "Third-Party Sync for repository {} has failed. Pollable Task ID: {}",
+        scheduledJob.getRepository().getName(),
+        pollableTaskId);
+
+    // Trigger PD incident
+    pagerDutyIntegrationService
+        .getDefaultPagerDutyClient()
+        .ifPresent(
+            pd -> {
+              String pollableTaskUrl =
+                  UriComponentsBuilder.fromHttpUrl(serverConfig.getUrl())
+                      .path("api/pollableTasks/" + pollableTaskId.toString())
+                      .build()
+                      .toUriString();
+
+              String scheduledJobUrl =
+                  UriComponentsBuilder.fromHttpUrl(serverConfig.getUrl())
+                      .path("api/jobs/" + scheduledJob.getId())
+                      .build()
+                      .toUriString();
+
+              String title =
+                  StrSubstitutor.replace(
+                      notificationTitle,
+                      ImmutableMap.of("repository", scheduledJob.getRepository().getName()),
+                      "{",
+                      "}");
+
+              PagerDutyPayload payload =
+                  new PagerDutyPayload(
+                      title,
+                      serverConfig.getUrl(),
+                      PagerDutyPayload.Severity.CRITICAL,
+                      ImmutableMap.of(
+                          "Pollable Task", pollableTaskUrl, "Scheduled Job", scheduledJobUrl));
+
+              try {
+                pd.triggerIncident(scheduledJob.getId(), payload);
+              } catch (PagerDutyException e) {
+                logger.error(
```

Comment:
You take a lot of time and effort to build up all those useful urls above and if the incident fails to trigger, then you just discard it? 
Can we not log it too to save us the trouble of reverse engineering the links? (It might be included in the payload, I'm not sure. Just the thought which was triggered)

Reply:
Done @byronantak